src/golang/lib/workflow/operator/base.go

```diff
@@ -76,6 +78,7 @@ func unknownSystemFailureExecState(err error, logMsg string) *shared.ExecutionSt
 }
 
 func (bo *baseOperator) launch(ctx context.Context, spec job.Spec) error {
+	bo.updateExecState(&shared.ExecutionState{Status: shared.RunningExecutionStatus})
```

Comment:
Can we add a check here that the current status must be `Pending`? I've see bugs that have caused infinite launch() loops before - this would make it immediately clear what's wrong.